app/src/main/java/com/gh4a/utils/RxUtils.java

```diff
@@ -0,0 +1,185 @@
+package com.gh4a.utils;
+
+import android.app.Dialog;
+import android.os.Bundle;
+import android.support.annotation.StringRes;
+import android.support.design.widget.CoordinatorLayout;
+import android.support.design.widget.Snackbar;
+import android.support.v4.app.DialogFragment;
+import android.support.v4.app.FragmentActivity;
+
+import com.gh4a.ApiRequestException;
+import com.gh4a.BaseActivity;
+import com.gh4a.R;
+import com.meisolsson.githubsdk.model.Page;
+import com.meisolsson.githubsdk.model.SearchPage;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.Comparator;
+import java.util.Iterator;
+import java.util.List;
+
+import io.reactivex.Single;
+import io.reactivex.SingleSource;
+import io.reactivex.SingleTransformer;
+import io.reactivex.android.schedulers.AndroidSchedulers;
+import io.reactivex.functions.Function;
+import io.reactivex.functions.Predicate;
+import io.reactivex.processors.PublishProcessor;
+import io.reactivex.schedulers.Schedulers;
+import retrofit2.Response;
+
+public class RxUtils {
+    public static <T> Single<T> throwOnFailure(Single<Response<T>> upstream) {
+        return upstream.map(response -> ApiHelpers.throwOnFailure(response));
+    }
+
+    public static <T> SingleTransformer<List<T>, List<T>> filter(Predicate<T> predicate) {
+        return upstream -> upstream.map(list -> {
+            if (list == null) {
+                return null;
+            }
+            Iterator<T> iter = list.iterator();
+            while (iter.hasNext()) {
+                if (!predicate.test(iter.next())) {
+                    iter.remove();
+                }
+            }
+            return list;
+        });
+    }
+
+    public static <T, R> SingleTransformer<List<T>, List<R>> mapList(Function<T, R> transformer) {
+        return upstream -> upstream.map(list -> {
+            if (list == null) {
+                return null;
+            }
+            List<R> result = new ArrayList<>();
+            for (T item : list) {
+                result.add(transformer.apply(item));
+            }
+            return result;
+        });
+    }
+
+    public static <T> SingleTransformer<List<T>, List<T>> sortList(Comparator<? super T> comparator) {
+        return upstream ->  upstream.map(list -> {
+            if (list != null) {
+                Collections.sort(list, comparator);
+            }
+            return list;
+        });
+    }
+
+    public static <T> SingleTransformer<T, T> mapFailureToValue(int code, T value) {
+        return upstream -> upstream.onErrorResumeNext(error -> {
+            if (error instanceof ApiRequestException) {
+                if (((ApiRequestException) error).getStatus() == code) {
+                    return Single.just(value);
+                }
+            }
+            return Single.error(error);
+        });
+    }
+
+    public static <T> Single<T> doInBackground(Single<T> upstream) {
+        return upstream.subscribeOn(Schedulers.newThread())
```

Comment:
Might be better to use `Schedulers.io()` or `Schedulers.computation()` to not create new thread every time.

Reply:
That one at least partly was intentional: we launched the AsyncTasks on the parallel executor before. Not sure what exactly AsyncTaskLoader does.